build_sysext

```diff
@@ -59,14 +73,30 @@ FLAGS "$@" || exit 1
 
 eval set -- "${FLAGS_ARGV}"
 
-source "${BUILD_LIBRARY_DIR}/board_options.sh" || exit 1
-source "${BUILD_LIBRARY_DIR}/reports_util.sh" || exit 1
+# Validate command line parameters
 
-if [[ -z "${FLAGS_build_dir}" ]]; then
-  die "Need a build directory to be specified with a --build_dir option"
+SYSEXTNAME="${1:-}"
+if [[ -z "${SYSEXTNAME}" ]]; then
+  die "No sysext name provided."
 fi
+shift
 
-BUILD_DIR=$(realpath "${FLAGS_build_dir}")
+if [[ -z "$FLAGS_squashfs_base" ]] ; then
+  FLAGS_squashfs_base="$(readlink -f "${SCRIPT_ROOT}/../build/images/${FLAGS_board}/latest/${FLATCAR_PRODUCTION_IMAGE_SYSEXT_BASE}")"
+fi
+if [[ ! -f "${FLAGS_squashfs_base}" ]] ; then
+  die "Squashfs base '${FLAGS_squashfs_base}' not found."
+fi
+
+if [[ -z "${FLAGS_image_builddir}" ]]; then
+  FLAGS_image_builddir="$(dirname "${FLAGS_squashfs_base}")/sysext"
+fi
+BUILD_DIR=$(realpath "${FLAGS_image_builddir}")
```

Comment:
Originally I thought we should be taking `BUILD_DIR` from `build_library/build_image_util.sh` but Krish ran into some problems.

Reply:
We dropped using `build_image_util.sh` at my request to replace that with just a build directory parameter, because OEM building is not using any of the output root, build attempts and other stuff that `build_image_util.sh` forces you to do. If someone wishes to use `build_image_util.sh`, he can do so by importing it by himself, and passing the `BUILD_DIR` to `build_sysext` through the `--build_dir` (or rather: `--image_builddir` now).